common/on_action/wc_aging_penalties.txt

```diff
@@ -0,0 +1,166 @@
+﻿on_birthday = {
+	on_actions = {
+		on_birthday_aging
+	}
+}
+on_birthday_aging = {
+	trigger = {
+		30 <= { value = age divide = racial_age_multiplier_value }
+	}
+
+	effect = {
+		if = {
+			limit = {
+				OR = {
+					has_trait_with_flag = no_aging_penalties
+					is_immortal = yes
+				}
+			}
+			clear_age_penalties = yes
+		}
+		else = {
+			save_temporary_scope_value_as = {
+				name = my_race_age_multiplier
+				value = racial_age_multiplier_value
+			}
+			
+			if = {
+				limit = {
+					aging_trigger_between = { AGE_1 = 30 AGE_2 = 35 }
+				}
+				clear_age_penalties = yes
+				add_age_penalty = { AGE = 30 }
+			}
+			else_if = {
+				limit = {
+					aging_trigger_between = { AGE_1 = 35 AGE_2 = 40 }
```

Comment:
Optimization.
If the effect fires, you're 35-40 years old, but do have the penalty modifier. This returns false and you go all the way down else/if chain.

Solution:
You split `aging_trigger_between` in 2 triggers: `has_aging_penalty_trigger = { AGE }` and `race_relative_age_between_trigger = { AGE_1 AGE_2 }`. First you check `race_relative_age_between_trigger`. If true, you check `has_aging_penalty_trigger`. If false, you get the modifier.